### PR TITLE
Brownfield apps dev tests

### PR DESCRIPTION
**Pull request by annakocheshkova**
Added full test coverage for file utils, update manager, package downloader and other classes.

### CODE REVIEW

File: android/app/src/androidTest/java/com/microsoft/codepush/common/AndroidTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        packageDownloader.setParameters(FULL_PACKAGE_URL, downloadFilePath, downloadProgressCallback);
+        return spy(packageDownloader);
+    }
+
+    /**
+     * Creates default spied package downloader instance.
+     *
+     * @return package downloader instance that can be mocked.
+     */
+    static PackageDownloader createPackageDownloader() {
```
Comment (ruslan-bikkinin) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/AndroidTestUtils.java]: Do we really need to combine specifics objects creation helpers in one file?

File: android/app/src/androidTest/java/com/microsoft/codepush/common/CodePushUpdateManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+        codePushUpdateManager.getCurrentPackageInfo();
+    }
+
+    /**
+     * Downloading files should return a {@link CodePushDownloadPackageException}
+     * if the amount of <code>receivedBytes</code> does not match the amount of <code>totalBytes</code>.
+     * The amount of <code>totalBytes</code> is more than <code>receivedBytes</code>.
+     */
+    @Test
+    public void downloadFailsIfBytesMismatchMore() throws Exception {
```
Comment (ruslan-bikkinin) [inline android/app/src/androidTest/java/com/microsoft/codepush/common/CodePushUpdateManagerTest.java]: General: `PackageDownloader` specific tests should be placed in separate file.

File: android/app/src/test/java/com/microsoft/codepush/common/CodePushConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+public class CodePushConfigurationTest {
+    @Before
+    public void setUp() {
+        mockStatic(AppCenterLog.class);
+    }
+
+    @Test
+    public void correctConfigurationTest() {
+        CodePushConfiguration correctConfig = new CodePushConfiguration();
+        correctConfig.setAppVersion(APP_VERSION)
```
Comment (ruslan-bikkinin) [inline android/app/src/test/java/com/microsoft/codepush/common/CodePushConfigurationTest.java]: It is a bad practice to share constant values between tests files. Each test should be isolated from each other. All constants references like `APP_VERSION`, `CLIENT_UNIQUE_ID` and so on should be replaced with hardcoded values. Please fix it here and in another test files.

File: android/app/src/test/java/com/microsoft/codepush/common/CodePushDataTest.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        assertEquals(CodePushInstallMode.IMMEDIATE, CodePushInstallMode.valueOf("IMMEDIATE"));
+        assertEquals(CodePushSyncStatus.AWAITING_USER_ACTION, CodePushSyncStatus.valueOf("AWAITING_USER_ACTION"));
+        assertEquals(CodePushCheckFrequency.MANUAL, CodePushCheckFrequency.valueOf("MANUAL"));
+    }
+
+    @Test
+    public void dataContractsTest() throws Exception {
+
+        /* Check download report. */
+        CodePushDownloadStatusReport codePushDownloadStatusReport = CodePushDownloadStatusReport.createReport(CLIENT_UNIQUE_ID, DEPLOYMENT_KEY, LABEL);
```
Comment (ruslan-bikkinin) [inline android/app/src/test/java/com/microsoft/codepush/common/CodePushDataTest.java]: Each datacontract should be tested in separate test file.

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,5 +1,6 @@
+#Fri Jan 26 16:45:56 MSK 2018
```
Comment (sergey-akhalkov) [inline android/gradle/wrapper/gradle-wrapper.properties]: Don't think if we should add this line.

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,5 +1,6 @@
...
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (sergey-akhalkov) [inline android/gradle/wrapper/gradle-wrapper.properties]: Do we really need to change gradle version? I would prefer to leave it as it is if possible.

File: android/build.gradle [hunk 7]
```diff
@@ -9,7 +9,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (sergey-akhalkov) [inline android/build.gradle]: Please note that this version of build gradle should be compatible with build tools 23.0.1

File: android/app/src/main/java/com/microsoft/codepush/common/utils/FileUtils.java [hunk 8]
```diff
@@ -189,6 +189,7 @@ public static String readFileToString(String filePath) throws IOException {
             Exception e = finalizeResources(
                     Arrays.asList(reader, fileInputStream),
                     "Error closing IO resources when reading file.");
+
             if (e != null) {
```
Comment (sergey-akhalkov) [inline android/app/src/main/java/com/microsoft/codepush/common/utils/FileUtils.java]: Add a comment above?

**Pull request merged**